### ISSUE TITLE
Network policy with range of ports is not handled correctly

### ISSUE THREAD

**Issue by Alexander-Bartosh** [Azure/AKS#2857]
Use-case: Allow egress from a range of ports.

Targeting a range of Ports 
https://kubernetes.io/docs/concepts/services-networking/network-policies/#targeting-a-range-of-ports

```
kind: NetworkPolicy
apiVersion: networking.k8s.io/v1
metadata:
  name: web-azuresql
spec:
  podSelector:
    matchLabels:
      networking/allow-db-access: 'true'
  egress:
    - ports:
        - protocol: TCP
          port: 11000
          endPort: 11999
```
        
**Environment**:
- Kubernetes version (use `kubectl version`): 1.22.6
- Network policy implementation: Azure Network Policies.

The policy above is created successfully. 
IP table rules are created only based on port. 
**endPort is ignored and thus egress is only allowed on 11000.** 

On a node: 
iptables -L | grep 11000: 
MARK       tcp  --  anywhere             anywhere             tcp dpt:11000 match-set azure-npm-482454837 src match-set azure-npm-2834923007 src /* ALLOW-ALL-T
O-TCP-PORT-11000-FROM-networking/allow-db-access:true-IN-ns-ns1 */ MARK or 0x1000

Is this expected behavior ? 
Was the support of Targeting a range of Ports implemented in Azure Network Policies? 
If not when it is planned to be implemented ? 

Thank you!

**Comment by vakalapa** [comment]
@Alexander-Bartosh Current NPM version in production does not support EndPort feature. A new version is being rolled out which supports this feature. Estimated rollout status of this new version to all regions is end of May. If this is a crucial feature for you to have, we can target an early release to your cluster or subscription. Please send an email to vakr@microsoft.com with subscription and cluster details for targeted early release. Thanks

**Comment by Alexander-Bartosh** [comment]
Thanks @vakalapa. 
I will wait for regular rollout  
In what NPM version is this fixed ? 
Currently running this one azure-npm:v1.4.21

**Comment by huntergregory** [comment]
The release is now scheduled to begin this coming Monday. Please reach out based on the below:

> If this is a crucial feature for you to have, we can target an early release to your cluster or subscription. Please send an email to [vakr@microsoft.com](mailto:vakr@microsoft.com) with subscription and cluster details for targeted early release. Thanks

**Comment by Mandur** [comment]
Thank you very much for the reply! We have a workaround for the moment so we will wait for the release.

**Comment by huntergregory** [comment]
Just noting that the fix for this was released in v1.4.32

**Comment by mllab-nl** [comment]
Can confirm, v1.4.32 creates a rage: 
tcp dpts:11000:11999 /* ALLOW-ALL-ON-TCP-TO-PORT-11000:11999 */

Thank you!
